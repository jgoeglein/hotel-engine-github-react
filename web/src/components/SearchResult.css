ul.search-results-list {
    list-style-type: none;
    width: 650px;
}

.search-result {
    padding: 12px;
    border-bottom: thin solid lightgray;
}

.search-result .stars, .search-result .language {
    color: gray;
    font-size: small;
}