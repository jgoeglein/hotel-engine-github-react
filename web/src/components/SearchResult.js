import React from 'react';
import './SearchResult.css';

class SearchResult extends React.PureComponent {

    render() {
        return <div className="search-result">
            <h3>
                <a href={this.props.result.html_url}>{this.props.result.name}</a> by {this.props.result.owner.login}
            </h3>
            <div className="description">{this.props.result.description}</div>
            <div className="language">Language: {this.props.result.language || 'Not specified'}</div>
            <div className="stars">{this.props.result.stargazers_count} stars</div>
        </div>
    }
}

export default SearchResult;