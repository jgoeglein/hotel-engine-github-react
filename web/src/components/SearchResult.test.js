import { cleanup, render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import SearchResult from './SearchResult';

afterEach(cleanup);

const testResult = {
  html_url: "fakeurl",
  description: "fake description",
  stargazers_count: 100,
  name: "fake repo name",
  language: "JS",
  owner: {
    login: "login"
  }
}

it('SearchResult renders without crashing', () => {
  const tree = renderer
    .create(<SearchResult result={testResult} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders Unspecified when language is not returned', () => {
  const nullLanguage = {
    ...testResult,
    language: null
  };
  const { queryByText } = render(
    <SearchResult result={nullLanguage} />
  );
  expect(queryByText(/Not specified/i)).toBeTruthy();
});