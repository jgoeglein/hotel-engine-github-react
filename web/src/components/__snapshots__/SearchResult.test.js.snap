// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult renders without crashing 1`] = `
<div
  className="search-result"
>
  <h3>
    <a
      href="fakeurl"
    >
      fake repo name
    </a>
     by 
    login
  </h3>
  <div
    className="description"
  >
    fake description
  </div>
  <div
    className="language"
  >
    Language: 
    JS
  </div>
  <div
    className="stars"
  >
    100
     stars
  </div>
</div>
`;
